shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformChannel.java

```diff
@@ -4,26 +4,536 @@
+     */
+    // TODO(mattcarroll): add @ScreenOrientation annotation
+    void setPreferredOrientations(int androidOrientation);
+
+    /**
+     * The Flutter application would like to be displayed in Android's app switcher with
+     * the visual representation described in the given {@code description}.
+     * <p>
+     * See the related Android documentation:
+     * https://developer.android.com/guide/components/activities/recents
+     */
+    void setApplicationSwitcherDescription(@NonNull AppSwitcherDescription description);
+
+    /**
+     * The Flutter application would like the Android system to display the given
+     * {@code overlays}.
+     * <p>
+     * {@link SystemUiOverlay#TOP_OVERLAYS} refers to system overlays such as the
+     * status bar, while {@link SystemUiOverlay#BOTTOM_OVERLAYS} refers to system
+     * overlays such as the back/home/recents navigation on the bottom of the screen.
+     * <p>
+     * An empty list of {@code overlays} should hide all system overlays.
+     */
+    void showSystemOverlays(@NonNull List<SystemUiOverlay> overlays);
+
+    /**
+     * The Flutter application would like to restore the visibility of system
+     * overlays to the last set of overlays sent via {@link #showSystemOverlays(List)}.
+     * <p>
+     * If {@link #showSystemOverlays(List)} has yet to be called, then a default
+     * system overlay appearance is desired:
+     * <p>
+     * {@code
+     * View.SYSTEM_UI_FLAG_LAYOUT_STABLE | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
+     * }
+     */
+    void restoreSystemUiOverlays();
+
+    /**
+     * The Flutter application would like the system chrome to present itself with
+     * the given {@code systemUiOverlayStyle}, i.e., the given status bar and
+     * navigation bar colors and brightness.
+     */
+    void setSystemUiOverlayStyle(@NonNull SystemChromeStyle systemUiOverlayStyle);
+
+    /**
+     * The Flutter application would like to pop the top item off of the Android
+     * app's navigation back stack.
+     */
+    void popSystemNavigator();
+
+    /**
+     * The Flutter application would like to receive the current data in the
+     * clipboard and have it returned in the given {@code format}.
+     */
+    @Nullable
+    CharSequence getClipboardData(@Nullable ClipboardContentFormat format);
+
+    /**
+     * The Flutter application would like to set the current data in the
+     * clipboard to the given {@code text}.
+     */
+    void setClipboardData(@NonNull String text);
+  }
+
+  /**
+   * Types of sounds the Android OS can play on behalf of an application.
+   */
+  public enum SoundType {
+    CLICK("SoundType.click");
+
+    static SoundType fromValue(@NonNull String encodedName) {
+      for (SoundType soundType : SoundType.values()) {
+        if (soundType.encodedName.equals(encodedName)) {
+          return soundType;
+        }
+      }
+      throw new RuntimeException("No such SoundType: " + encodedName);
```

Comment:
Subjective: Is throwing an exception/crashing the app really the right thing to do here? Should this be checked instead of unchecked? https://stackoverflow.com/a/3540705

In general it's subjective but we probably don't want to throw an actual top level `RuntimeException` specifically. Are any of its subclasses better fits for this? Maybe `JsonError`? Technically this is a String, but really this is a downstream JSON parsing error.

This should also probably be documented somewhere even if it's unchecked so that callers know they're risking a thrown exception when they call it.

This applies to all the "fromValue" methods.

Reply:
I went with NoSuchFieldException for each enum lookup method.  I also wrapped every respective message handler with a try/catch for both JSONException and NoSuchFieldException as needed.